test/behavior/slice.zig

```diff
@@ -584,6 +585,14 @@ test "slice syntax resulting in pointer-to-array" {
             try comptime expect(@TypeOf(ptr_z[1.. :0][0..4]) == *[4]u8);
             try comptime expect(@TypeOf(ptr_z[1.. :0][0..2 :4]) == *[2:4]u8);
         }
+
+        fn testSingleItemPointer() !void {
+            var value: u8 = 1;
+            var ptr = &value;
+            const slice = ptr[0..1];
+            try comptime expect(@TypeOf(slice) == *[1]u8);
```

Comment:
For checking the type, `comptime assert` is generally preferred, so that we get a compile error rather than a runtime test failure (or unhelpful later compile error) - doesn't hugely matter though

Reply:
> For checking the type, `comptime assert` is generally preferred...

There is a mix of `try comptime expect(...)` and `comptime assert(...)` in `test/behavior` - going to stick with the `try comptime expect(...)` pattern that `test/behavior/slice.zig` currently uses.